hypex/dataset/abstract.py

```diff
@@ -46,21 +46,12 @@ def _set_all_roles(self, roles):
         return roles
 
     def _set_empty_types(self, roles):
-        types_map = {
-            "int": int,
-            "float": float,
-            "object": str,
-            "bool": bool,
-            "category": str,
-        }
-        reversed_map = {int: "int", float: "float", str: "category", bool: "bool"}
         for column, role in roles.items():
             if role.data_type is None:
-                d_type = self.backend.get_column_type(column)
-
-                role.data_type = [v for k, v in types_map.items() if k in d_type][0]
-            self._backend = self.backend.update_column_type(
-                column, reversed_map[role.data_type]
```

Comment:
Why has this check been removed? What if pandas dataframe has non-suspported data type?

Reply:
I don't see these checks added to the backend/abstract.py.

Reply:
OK, found it, but it is not clear why were the checks moved. Also the new implementation is less lean.